assets/javascripts/discourse/initializers/discourse-reactions.gjs

```diff
@@ -220,13 +177,93 @@ function initializeDiscourseReactions(api) {
   }
 }
 
+function customizePostMenu(api, container) {
+  const currentUser = container.lookup("service:current-user");
+
+  const transformerRegistered =
+    currentUser?.use_glimmer_post_menu &&
+    api.registerValueTransformer(
+      "post-menu-registered-buttons",
+      ({ value: dag }) => {
+        dag.replace(POST_MENU_LIKE_BUTTON_KEY, ReactionsActionButton);
+        dag.add("discourse-reactions-actions", ReactionsActionSummary, {
+          after: POST_MENU_REPLIES_BUTTON_KEY,
+        });
+
+        return dag;
+      }
+    );
+
+  const silencedKey =
+    transformerRegistered && "discourse.post-menu-widget-overrides";
+
+  withSilencedDeprecations(silencedKey, () => customizeWidgetPostMenu(api));
```

Comment:
Cool! We should make sure to document this pattern.

Does silencing the deprecation also stop it from affecting the 'glimmer_post_menu_mode=auto' system?

Reply:
Yep. Silencing the deprecations stops them triggering the legacy mode.

I guess introducing a new deprecation may be a good idea on this case. Or we can log an error message in the console like we did in the header.